### PR TITLE
Fix KaaS teams

### PR DESCRIPTION
**Pull request by tfussell**
Address root cause of https://github.com/giantswarm/giantswarm/issues/20460

This PR:

- Send SE alerts to correct KaaS teams.
- Use a template definition for KaaS team templating.

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).
- [x] Alerting rules must have a comment documenting why it needs to exist.

### CODE REVIEW

File: helm/prometheus-rules/templates/_helpers.tpl [hunk 1]
```diff
@@ -26,3 +26,11 @@ app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
 helm.sh/chart: {{ include "chart" . | quote }}
 giantswarm.io/service-type: {{ .Values.serviceType }}
 {{- end -}}
+
+{{- define "providerTeam" -}}
+{{- if has .Values.managementCluster.provider.kind (list "kvm" "vsphere" "openstack") -}}
```
Comment (tfussell) [inline helm/prometheus-rules/templates/_helpers.tpl:31]: Allows us to easily adjust provider<>team mapping in the future.

**Comment by tfussell** [comment]
Not sure if anyone care to look, but here is the diff between provider being set to "aws" and then "kvm": [diff.txt](https://github.com/giantswarm/prometheus-rules/files/7919169/diff.txt)

**Review by anvddriesch — APPROVED** [review]

**Review by AverageMarcus — APPROVED** [review]

**Review by njuettner — APPROVED** [review]

**Review by yulianedyalkova — APPROVED** [review]

**Review by pipo02mix — APPROVED** [review]
nice

File: helm/prometheus-rules/templates/alerting-rules/aws.management-cluster.rules.yml [hunk 2]
```diff
@@ -103,7 +103,7 @@ spec:
       labels:
         area: kaas
         severity: notify
-        team: se
+        team: phoenix
```
Comment (whites11) [inline helm/prometheus-rules/templates/alerting-rules/aws.management-cluster.rules.yml:106]: Why aren't you using `{{ include "providerTeam" . }}` here as well?

File: helm/prometheus-rules/templates/alerting-rules/aws.management-cluster.rules.yml [hunk 3]
```diff
@@ -114,7 +114,7 @@ spec:
...
```
Comment (whites11) [inline helm/prometheus-rules/templates/alerting-rules/aws.management-cluster.rules.yml:117]: Why aren't you using `{{ include "providerTeam" . }}` here as well?

File: helm/prometheus-rules/templates/alerting-rules/azure.management-cluster.rules.yml [hunk 4]
```diff
@@ -258,7 +258,7 @@ spec:
       labels:
         area: kaas
         severity: notify
-        team: se
+        team: phoenix
```
Comment (whites11) [inline helm/prometheus-rules/templates/alerting-rules/azure.management-cluster.rules.yml:261]: Why aren't you using `{{ include "providerTeam" . }}` here as well?

File: helm/prometheus-rules/templates/alerting-rules/aws.management-cluster.rules.yml (hunk 2 above)
Comment (AverageMarcus) [reply helm/prometheus-rules/templates/alerting-rules/aws.management-cluster.rules.yml:106]: These are phoenix specific files (aws and azure) but I think it'd be worth using the include here too as in the future changing responsibilities for different clouds is easier.

Comment (tfussell) [reply helm/prometheus-rules/templates/alerting-rules/aws.management-cluster.rules.yml]: I've changed these three that you pointed out. We still hardcode phoenix and rocket in almost 200 rules, but the chance of breaking things is too high to do all of them so I'm going to restrict to these changes (`team: se` and existing provider specific team assignment) for now.

**Review by whites11 — APPROVED** [review]

**Pull request merged**